File: Codeforces/1462A-Favorite_Sequence.py
def solve(arr, n):
    ans = [0] * n
    left, right = 0, n - 1
    i = 0

    while i < n:
        ans[i] = arr[left]
        i += 1
        if i < n:
            ans[i] = arr[right]
            i += 1
        left += 1
        right -= 1
    return ans

for _ in range(int(input())):
    n = int(input())
    arr = list(map(int, input().split()))

    ans = solve(arr, n)
    print(*ans)

File: Codeforces/1462A-Favorite_Sequence.c
#include <stdio.h>
#include <stdlib.h>

int* solve(int arr[], int n) {
  int *ans = (int*) malloc(n * sizeof(int));
  int left = 0, right = n - 1, i = 0;

  while (i < n) {
    ans[i++] = arr[left];
    if (i < n) ans[i++] = arr[right];

    left++; right--;
  }
  return ans
}

int main() {
  int t;
  scanf("%d", &t);

  while (t--) {
    int n;
    scanf("%d", &n);

    int arr[n];
    for (int i = 0; i < n; i++) {
      scanf("%d", &arr[i]);
    }

    int *ans = solve(arr, n);
    for (int i = 0; i < n; i++) {
      printf("%d ", ans[i]);
    }

    printf("\n");
    free(ans);
  }
}
